App/Components/Chat/CompChatView.razor

```diff
@@ -168,73 +269,152 @@
 @* ReSharper disable always InconsistentNaming *@
 @code {
     [Inject] IUserGroupService UserGroupService { get; set; } = default!;
-    
+
     [CascadingParameter] public EventCallback<KnownPopupArgs> OpenKnownPopup { get; set; }
-    
-    [Parameter] public int? ChatID { get; set; } 
-    public List<int?> WatchedIDs { get; set; } = [];
+    [CascadingParameter] public EventCallback ClosePopupCallback { get; set; }
+    [CascadingParameter] public EventCallback<IPacket> SendToServer { get; set; }
+    [CascadingParameter] public Predicate<IAsyncPacketSubscriber> AddPacketSubscriber { get; set; }
+    [CascadingParameter] public Predicate<IAsyncKeySubscriber> AddKeySubscriber { get; set; }
+
+    [Parameter] public int? ChatID { get; set; }
     private Chat? Chat { get; set; }
+    private string ChatName { get; set; }
+    private bool ChatLoaded { get; set; }
+
+    private User? _otherUser;
+
+    /// <summary>
+    ///     Null if the chat is a group chat
+    /// </summary>
+    private User? OtherUser
+    {
+        get
+        {
+            if (Chat is GroupChat) return null;
+
+            return _otherUser ??= Chat?.Users.FirstOrDefault(u => u.Id != Self!.Id);
+        }
+        set => _otherUser = value;
+    }
+
+    public List<int?> WatchedIDs { get; private set; } = [];
+
     private User? Self { get; set; }
-    private string ChatName { get; set; } = default!;
     private bool IsMod { get; set; }
-    private List<LinkedMessage> Messages { get; set; } = [];
-    
-    private DateTime? _lastMessageSentAt;
-    
-    private string MessageInput { get; set; } = default!;
 
-    private string _hubUrl = default!;
-    private HubConnection? _hubConnection;
+    private List<MessageEntry> Messages { get; set; } = [];
```

Comment:
Messages, EphemeralMessages & MessageIdToIndex are not cleared out when the chat is changed, so the same messages are taken to each different chat display :(